connectors/tests/test_es_document.py

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+import pytest
+
+from connectors.es import ESDocument, InvalidDocumentSourceError
+
+
+@pytest.mark.parametrize(
+    "doc_source, exception",
+    [
+        (None, InvalidDocumentSourceError),
+        ("hahaha", InvalidDocumentSourceError),
+        ({}, InvalidDocumentSourceError),
+        ({"_id": {}}, InvalidDocumentSourceError),
+        ({"_id": "1", "_source": "hahaha"}, InvalidDocumentSourceError),
+        ({"_id": "1", "_source": {}}, None),
+    ],
+)
+def test_es_document(doc_source, exception, patch_logger):
+    try:
+        ESDocument(elastic_index=None, doc_source=doc_source)
+    except Exception as e:
```

Comment:
look at `pytest.raises` https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

Reply:
Yes, I planned to use `pytest.raises` initially, but there are test cases which don't raise errors, and `pytest.raises` won't work.

Then I say this: https://miguendes.me/how-to-check-if-an-exception-is-raised-or-not-with-pytest#heading-how-to-assert-that-no-exception-is-raised, and I think this pattern works well for this particular test.

Reply:
> there are test cases which don't raise errors, and pytest.raises won't work.

if you don't want to assert that an error was not raised, don't use `pytest.raises` for sure, the problem is your attempt to parameterized even when it's not raising..

here's your test fixed (untested):

```
@pytest.mark.parametrize(
    "doc_source, exception",
    [None,  "hahaha", {}, {"_id": {}}, {"_id": "1", "_source": "hahaha"}] 
)
def test_es_document_raise(doc_source, patch_logger):
    with pytest.raises(InvalidDocumentSourceError):
        ESDocument(elastic_index=None, doc_source=doc_source)

def test_es_document_ok():
    doc = ESDocument(elastic_index=None, doc_source={"_id": "1", "_source": {}})
    assert something on doc
```

Reply:
Done here: https://github.com/elastic/connectors-python/pull/460/commits/b2b1060798ce2f2e401951b609d49870c09fd5e2